numpy/core/src/multiarray/getset.c

```diff
@@ -419,6 +419,29 @@ array_nbytes_get(PyArrayObject *self)
 #endif
 }
 
+static int
+_use_C_view_dimension(PyArrayObject *ap)
+{
+    npy_intp sd;
+    npy_intp dim;
+    int i;
+    npy_bool is_c_contig = 1;
+
+    sd = PyArray_ITEMSIZE(ap);
+    for (i = PyArray_NDIM(ap) - 1; i >= 0; --i) {
+        dim = PyArray_DIMS(ap)[i];
```

Comment:
There is a `PyArray_DIM` function that you can use to get a single dimension of an array, i.e. `PyArray_DIM(ap, i)` in this case.  That function does the exact same thing you are doing, but it is probably better to use our own API.